pkg/chains/storage/grafeas/grafeas_test.go

```diff
@@ -0,0 +1,350 @@
+											Signature: []byte("taskrun signature"),
+											// PublicKeyId: we're only using KMS for signing which is the one we currently set its reference in attestation
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Envelope: &commonpb.Envelope{
+					Payload:     []byte("taskrun payload"),
+					PayloadType: "in-toto attestations containing a slsa.dev/provenance predicate",
+					Signatures: []*commonpb.EnvelopeSignature{
+						{
+							Sig: []byte("taskrun signature"),
+							// PublicKeyId: we're only using KMS for signing which is the one we currently support for storing its reference in attestation
+						},
+					},
+				},
+			},
+			// occurrence for OCI image
+			{
+				Name:     "gcr.io/test/kaniko-chains1@sha256:cfe4f0bf41c80609214f9b8ec0408b1afb28b3ced343b944aaa05d47caba3e00",
+				Resource: &pb.Resource{Uri: "gcr.io/test/kaniko-chains1@sha256:cfe4f0bf41c80609214f9b8ec0408b1afb28b3ced343b944aaa05d47caba3e00"},
+				NoteName: "projects/test-project/notes/test-note",
+				Details: &pb.Occurrence_Attestation{
+					Attestation: &attestationpb.Details{
+						Attestation: &attestationpb.Attestation{
+							Signature: &attestationpb.Attestation_GenericSignedAttestation{
+								GenericSignedAttestation: &attestationpb.GenericSignedAttestation{
+									ContentType:       attestationpb.GenericSignedAttestation_SIMPLE_SIGNING_JSON,
+									SerializedPayload: []byte("oci payload"),
+									Signatures: []*commonpb.Signature{
+										{
+											Signature: []byte("oci signature"),
+											// PublicKeyId: we're only using KMS for signing which is the one we currently support for storing its reference in attestation
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Envelope: &commonpb.Envelope{
+					Payload:     []byte("oci payload"),
+					PayloadType: "in-toto attestations containing a slsa.dev/provenance predicate",
+					Signatures: []*commonpb.EnvelopeSignature{
+						{
+							Sig: []byte("oci signature"),
+							// PublicKeyId: we're only using KMS for signing which is the one we currently support for storing its reference in attestation
+						},
+					},
+				},
+			},
+		},
+	}
+	got, err := b.client.ListOccurrences(ctx,
+		&pb.ListOccurrencesRequest{
+			// This is just a placeholder.
+			// ProjectID doesn't matter here because we assume there is only one project in the mocked server.
+			Parent: fmt.Sprintf("project/%s", b.cfg.Storage.Grafeas.ProjectID),
+		},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// set compare option to ignore all unexported fields of ListOccurrencesResponse struct
+	// including unexported fields of the embedded structs.
+	opt := cmpopts.IgnoreUnexported(
+		pb.ListOccurrencesResponse{},
+		pb.Occurrence{},
+		pb.Resource{},
+		attestationpb.Details{},
+		attestationpb.Attestation{},
+		attestationpb.GenericSignedAttestation{},
+		commonpb.Signature{},
+		commonpb.Envelope{},
+		commonpb.EnvelopeSignature{},
+	)
```

Comment:
Typically it's not common to need to ignore all of these individually - usually I see a single [protocmp.Transform()](https://pkg.go.dev/google.golang.org/protobuf/testing/protocmp#Transform) is used instead (unless individual fields need to be ignored). Can this be used instead?

Reply:
wow this is a great improvement. It works!